src/_pytest/fixtures.py

```diff
@@ -950,6 +950,8 @@ def pytest_fixture_setup(fixturedef, request):
         kwargs[argname] = result
 
     fixturefunc = resolve_fixture_function(fixturedef, request)
+    if hasattr(fixturefunc,'pytestmark'):
+        raise Exception('i did it!')
```

Comment:
Use `fail(msg, pytrace=False)` instead. 😁 

Also this should go into `features`, as it might break functional (but incorrect) test suites.

Reply:
oh thanks @nicoddemus.
so the course of action will be:
+ move to `features`.
+ correct the error invokation.
+ write tests for this.
+ then work on the other half
right?

Reply:
Correct. Thanks for tackling this! 👍